### ISSUE TITLE
Build libspdk failed on arm64

### ISSUE THREAD

**Issue by xin3liang** [openebs/mayastor#1080]
**Describe the bug**
Build libspdk failed on arm64

**To Reproduce**
```
ubuntu@test-focal:~/work/k8s/openebs/mayastor$ nix develop -f . mayastor --extra-experimental-features nix-command
error: builder for '/nix/store/1a6lh7j3mfy16mk67x5l4rnw2rrnzvpp-libspdk-21.10-8709a5789.drv' failed with exit code 1;
       last 10 log lines:
       > --------------------------
       > |diff --git a/intel-ipsec-mb/lib/Makefile b/intel-ipsec-mb/lib/Makefile
       > |index ddb4cb4..61dadbf 100644
       > |--- a/intel-ipsec-mb/lib/Makefile
       > |+++ b/intel-ipsec-mb/lib/Makefile
       > --------------------------
       > File to patch:
       > Skip this patch? [y]
       > Skipping patch.
       > 1 out of 1 hunk ignored
       For full logs, run 'nix log /nix/store/1a6lh7j3mfy16mk67x5l4rnw2rrnzvpp-libspdk-21.10-8709a5789.drv'.
error: 1 dependencies of derivation '/nix/store/q8mbdyzdwwkib6b3dw133hkn998h38jc-mayastor-env.drv' failed to build

```

**Expected behavior**
Success

**Screenshots**

** OS info (please complete the following information):**
 - Distro: ubuntu-focal
 - Kernel version: 5.4.0-96-generic aarch64
 - MayaStor revision or container image: develop branch revision b3247777dfecf836eb4e56a3b9422c0d583bc078

**Comment by tiagolobocastro** [comment]
@xin3liang AFAIK we don't have flakes enabled so you'd have to run nix-shell instead, eg: `nix-shell ci.nix`.

**Comment by jonathan-teh** [comment]
This specific build issue is fixed by #1087 but there could be other issues as arm64 is not a supported platform.

**Comment by xin3liang** [comment]
> @xin3liang AFAIK we don't have flakes enabled so you'd have to run nix-shell instead, eg: `nix-shell ci.nix`. You can then try to run `cargo build --bin mayastor --release`. Alternatively you can try to build the docker image direcly (run outside of nix-shell): `nix-build -A images.mayastor`


Thanks @tiagolobocastro's notices. will try.

**Comment by xin3liang** [comment]
> This specific build issue is fixed by #1087 but there could be other issues as arm64 is not a supported platform.

Verified that this issue is fixed. Thanks @jonathan-teh. 
Yes, arm64 platform is not support yet. There is still other issues to be fix for compiling. One issue is that SPDK didn't compile crypto driver for arm64. Crypto lib seems to be required for [spdk-rs](https://github.com/mayadata-io/spdk-rs/blob/5f3cd2bdb173da1484554d283e51b37a20074a73/build.rs#L138). But when I remove [this line](https://github.com/mayadata-io/spdk-rs/blob/5f3cd2bdb173da1484554d283e51b37a20074a73/build.rs#L138)，it still can be compiled. 
FYI, See my hacks to build SPDK: https://github.com/xin3liang/spdk/tree/v22.01.x-mayastor-fix-spdk-dpdk-build

**Comment by xin3liang** [comment]
This issue can be closed. As the issue is fixed.
